plugins/goodreads.koplugin/goodreadsapi.lua

```diff
@@ -0,0 +1,190 @@
+    request['sink'] = ltn12.sink.table(sink)
+    http.TIMEOUT = 5
+    https.TIMEOUT = 5
+    local httpRequest = parsed.scheme == 'http' and http.request or https.request
+    local headers = socket.skip(1, httpRequest(request))
+    if headers == nil then
+        return nil
+    end
+    local xml = table.concat(sink)
+    if xml ~= "" then
+        return xml
+    end
+end
+
+function GoodReadsApi:showSearchTable(data)
+    local books = {}
+    if data == nil then
+        UIManager:show(InfoMessage:new{text =_("Network problem.\nCheck connection.")})
+        return {}
+    end
+    self.total_result = data:match("<total[-]results>(.*)</total[-]results>")
+
+    for work in data:gmatch("<work>(.-)</work>") do
+        local book = work:match("<best_book[^>]+>(.*)</best_book>")
+        local id = book:match("<id[^>]+>([^<]+)</id>")
+        local title = book:match("<title>([^<]+)</title>"):gsub(" %(.*, #%d+%)$", "")
+        local author = book:match("<name>([^<]+)</name>")
+        table.insert(books, {
+            author = author,
+            title = title,
+            id = id,
+        })
+    end
+    if #books == 0 then
+        UIManager:show(InfoMessage:new{text =_("Search not found!")})
+    end
+    return books
+end
+
+function GoodReadsApi:getTotalResults()
+    return self.total_result
+end
+
+local function showIdTable(data)
+    if data == nil then
+        UIManager:show(InfoMessage:new{text =_("Network problem.\nCheck connection.")})
+        return {}
+    end
+    local data1 = data:match("<book>(.*)</reviews_widget>")
+    local title_all = data1:match("<title>(.*)</title>"):gsub("<![[]CDATA[[]", ""):gsub("]]>$", "")
+    local title = title_all:gsub(" %(.*, #%d+%)$", "")
+    local average_rating = data1:match("<average_rating>([^<]+)</average_rating>")
+    local series = title_all:match("%(.*, #%d+%)$")
+    if series ~= nil then
+        series = series:match("[(](.*)[)]")
+    else
+        series = _("N/A")
+    end
+    local num_pages = data1:match("<num_pages>(.*)</num_pages>"):gsub("<![[]CDATA[[]", ""):gsub("]]>$", "")
+    if num_pages == nil or num_pages =="" then
+        num_pages = _("N/A")
+    end
+    local id = data1:match("<id>([^<]+)</id>"):gsub("<![[]CDATA[[]", ""):gsub("]]>$", "")
+    local author = data1:match("<name>([^<]+)</name>")
+    local description = data1:match("<description>(.*)</description>")
+    description = description:gsub("<![[]CDATA[[]", ""):gsub("]]>$", ""):gsub("<br>", "")
+    --change format from medium to large
+    local image = data1:match("<image_url>([^<]+)</image_url>"):gsub("([0-9]+)m/", "%1l/")
+    local day = data1:match("<original_publication_day[^>]+>([^<]+)</original_publication_day>")
+    local month = data1:match("<original_publication_month[^>]+>([^<]+)</original_publication_month>")
+    local year = data1:match("<original_publication_year[^>]+>([^<]+)</original_publication_year>")
+
+    local release = {}
+    if(year) then
```

Comment:
nitpick, please keep the coding style consistent, i.e. keep one space after `if`.